src/node_api.cc

```diff
@@ -0,0 +1,2430 @@
+      "Count of error messages must match count of error values");
+  assert(static_last_error.error_code < napi_status_last);
+
+  // Wait until someone requests the last error information to fetch the error
+  // message string
+  static_last_error.error_message =
+      error_messages[static_last_error.error_code];
+
+  return &static_last_error;
+}
+
+napi_status napi_set_last_error(napi_status error_code,
+                                uint32_t engine_error_code = 0,
+                                void* engine_reserved = nullptr) {
+  static_last_error.error_code = error_code;
+  static_last_error.engine_error_code = engine_error_code;
+  static_last_error.engine_reserved = engine_reserved;
+
+  return error_code;
+}
+
+napi_status napi_create_function(napi_env e,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* data,
+                                 napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  v8::Isolate *isolate = v8impl::V8IsolateFromJsEnv(e);
+  v8::Local<v8::Function> retval;
+
+  v8::EscapableHandleScope scope(isolate);
+
+  v8::Local<v8::Object> cbdata =
+      v8impl::CreateFunctionCallbackData(e, cb, data);
+
+  RETURN_STATUS_IF_FALSE(!cbdata.IsEmpty(), napi_generic_failure);
+
+  v8::Local<v8::FunctionTemplate> tpl = v8::FunctionTemplate::New(
+      isolate, v8impl::FunctionCallbackWrapper::Invoke, cbdata);
+
+  retval = scope.Escape(tpl->GetFunction());
+
+  if (utf8name) {
+    v8::Local<v8::String> namestring;
+    CHECK_NEW_FROM_UTF8(isolate, namestring, utf8name);
+    retval->SetName(namestring);
+  }
+
+  *result = v8impl::JsValueFromV8LocalValue(retval);
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_define_class(napi_env e,
+                              const char* utf8name,
+                              napi_callback constructor,
+                              void* data,
+                              int property_count,
+                              const napi_property_descriptor* properties,
+                              napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  v8::Isolate* isolate = v8impl::V8IsolateFromJsEnv(e);
+
+  v8::EscapableHandleScope scope(isolate);
+  v8::Local<v8::Object> cbdata =
+      v8impl::CreateFunctionCallbackData(e, constructor, data);
+
+  RETURN_STATUS_IF_FALSE(!cbdata.IsEmpty(), napi_generic_failure);
+
+  v8::Local<v8::FunctionTemplate> tpl = v8::FunctionTemplate::New(
+      isolate, v8impl::FunctionCallbackWrapper::Invoke, cbdata);
+
+  // we need an internal field to stash the wrapped object
+  tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+  v8::Local<v8::String> namestring;
+  CHECK_NEW_FROM_UTF8(isolate, namestring, utf8name);
+  tpl->SetClassName(namestring);
+
+  int staticPropertyCount = 0;
+  for (int i = 0; i < property_count; i++) {
```

Comment:
Resolved by https://github.com/nodejs/node/pull/11975/commits/9e3ab8317b107ecd4456aecf302d7ed4fb8af2e9 (`size_t`) and https://github.com/nodejs/node/pull/11975/commits/fb8ced493fe5313a2fd082bd6f23d24b268b644f (snake_case)